### PR TITLE
feat(core): new built-in `maskitoScrollPlugin`

### LINKED ISSUE #917: 🐞 - When input is narrow and value doesn't fit into input, setSelectionRange don't scroll to cursor when typing
### Which package(s) are the source of the bug?

@maskito/core

### Playground Link

https://codesandbox.io/p/sandbox/quiet-wildflower-6ffc5y?file=%2Fsrc%2FApp.tsx%3A58%2C23

### Description

1. Open https://codesandbox.io/p/sandbox/quiet-wildflower-6ffc5y?file=%2Fsrc%2FApp.tsx%3A58%2C23
2. Type 123456789 into Maskito input
3. Last symbols should be visible (456.789 in this example), but setSelectionRange doesn't scroll to cursor position (in firefox something another causes this behavior)

### Maskito version

1.9.0

### Which browsers have you used?

- [X] Chrome
- [X] Firefox
- [ ] Safari
- [X] Edge

### Which operating systems have you used?

- [x] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by nsbarsukov**
## Previous behavior

https://github.com/user-attachments/assets/0e6a603a-e2e5-4729-bab8-aae9918871b4

## New behavior

https://github.com/user-attachments/assets/4a0d976f-c310-49bd-977c-84f655ad833d

Closes #917

### CODE REVIEW

File: projects/core/src/lib/plugins/scroll-plugin.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * input.setSelectionRange(5, 5);
+ * // caret is placed after the last digit, but it is not visible (no scroll happens)
+ * ```
+ * ___
+ * - {@link https://issues.chromium.org/issues/41081857 Open Chromium bug from 2014}
+ */
+export function maskitoScrollPlugin(): MaskitoPlugin {
+    return (element) => {
+        const document = element.ownerDocument;
```
Comment (splincode) [inline projects/core/src/lib/plugins/scroll-plugin.ts:34]: ssr friendly?

Comment (nsbarsukov) [reply projects/core/src/lib/plugins/scroll-plugin.ts:34]: yes

File: projects/core/src/lib/plugins/scroll-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        const listener = (): void => {
+            if (element.matches(':focus') && element.scrollWidth > element.clientWidth) {
+                element.blur();
+                element.focus();
```
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/core/src/lib/plugins/scroll-plugin.ts:39]: alternatively it won't work

[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/core/src/lib/plugins/scroll-plugin.ts:39]: https://stackoverflow.com/a/40951875

[the remark this answers is not in the document]
Comment (waterplea) [reply projects/core/src/lib/plugins/scroll-plugin.ts:39]: We cannot just call `blur`, it will close dropdowns in Taiga UI if input is in one

**Pull request closed without merging**